app/src/main/java/com/example/composetemplate/repositories/FirebaseActionable.kt

```diff
@@ -1,89 +1,139 @@
 package com.example.composetemplate.repositories
 
 import android.util.Base64
+import androidx.credentials.CredentialManager
+import androidx.credentials.GetCredentialRequest
+import androidx.credentials.exceptions.GetCredentialCancellationException
+import androidx.credentials.exceptions.GetCredentialException
+import androidx.credentials.exceptions.NoCredentialException
+import com.example.composetemplate.data.models.local_models.ErrorType
+import com.example.composetemplate.data.models.local_models.GoogleCredentialAuthParameter
 import com.example.composetemplate.data.models.local_models.User
+import com.example.composetemplate.data.models.server_models.PermissionType
+import com.example.composetemplate.data.remote.confing.FirebaseConfigProvider
+import com.example.composetemplate.data.remote.confing.remoteConfigVal
+import com.example.composetemplate.data.remote.errors.AuthError
 import com.example.composetemplate.managers.TokenData
 import com.example.composetemplate.managers.TokenFetcher
-import com.example.composetemplate.utils.LoginCallback
-import com.example.composetemplate.utils.SuccessCallback
+import com.example.composetemplate.presentation.screens.entry_screens.login.SignInResult
+import com.example.composetemplate.presentation.screens.entry_screens.login.SignUpResult
+import com.example.composetemplate.utils.extensions.errorType
+import com.google.android.gms.auth.api.identity.SignInClient
+import com.google.android.libraries.identity.googleid.GetGoogleIdOption
+import com.google.android.libraries.identity.googleid.GoogleIdTokenCredential
 import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
 import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.tasks.await
 import org.json.JSONObject
+import java.security.MessageDigest
+import java.util.UUID
 import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
 
 /**
- * In the MVVM architecture, the DataSource layer is responsible for providing data to the repository
- * from various sources, such as remote APIs, local databases, or in-memory caches.
- * allowing the repository to seamlessly aggregate and supply data to the ViewModel without worrying about the underlying data origins or access mechanisms.
+ * The FirebaseActionable class handles all Firebase authentication tasks.
+ * It manages user sign-up, sign-in, password reset, and token fetching using Firebase Authentication SDK.
+ * This class also implements TokenFetcher to retrieve and manage the Firebase token.
  */
 class FirebaseActionable : AuthActionable, TokenFetcher {
```

Comment:
I think we should think about better naming for this class. It's naming of interface and it's a class. Maybe FirebaseActionsManager? you can ask chatGPT.

Reply:
I will change it to FirebaseAuthManager because we have only auth actions here. other firebase action will be in seperate manager